addons/mail/static/src/core/common/thread_model.js

```diff
@@ -41,7 +41,19 @@ export class Thread extends Record {
         return super.insert(...arguments);
     }
     static async getOrFetch(data) {
-        return this.get(data);
+        let thread = this.get(data);
+        if (thread?.displayName === undefined) {
```

Comment:
Note: arbitrary field name to determine if we need to fetch (which is actually displayed in chat window).

We should probably consider giving actual fields to fetch as method param.